src/t8_geometry/t8_geometry_implementations/t8_geometry_occ.cxx

```diff
@@ -0,0 +1,242 @@
+
+Handle_Geom_Curve t8_global_occ_curve[10];
+Handle_Geom_Surface t8_global_occ_surface[10];
+
+t8_geometry_occ::t8_geometry_occ (int dim, const char *name_in,
+                                  const void *user_data_in)
+{
+  T8_ASSERT (0 <= dim && dim <= 3);
+
+  name = name_in;
+  dimension = dim;
+  user_data = user_data_in;
+}
+
+/**
+ * Map a point in the reference space $$[0,1]^dimension$$ to $$\mathbb R^3$$
+ * \param [in]  gtreeid     The global tree (of the cmesh) in which the reference point is.
+ * \param [in]  ref_coords  Array of \a dimension many entries, specifying a point in [0,1]^dimension.
+ * \param [out] out_coords  The mapped coordinates in physical space of \a ref_coords.
+ */
+void
+t8_geometry_occ::t8_geom_evaluate (t8_cmesh_t cmesh,
+                                  t8_gloidx_t gtreeid,
+                                  const double *ref_coords,
+                                  double out_coords[3]) const
+{
+  const int *edges = (const int *) t8_cmesh_get_attribute (cmesh, t8_get_package_id (),
+                                                          T8_CMESH_OCC_CURVE_ATTRIBUTE_KEY,
+                                                          gtreeid);
+  const int *faces = (const int *) t8_cmesh_get_attribute (cmesh, t8_get_package_id (),
+                                                          T8_CMESH_OCC_SURFACE_ATTRIBUTE_KEY,
+                                                          gtreeid);
+  
+  /* Compute coordinates via trilinear interpolation */
+  t8_geom_compute_linear_geometry (active_tree_class,
+                                   active_tree_vertices, ref_coords,
+                                   out_coords);
+  
+  /* Sequence of normal vectors to do linear interpolations */
+  int normal_sequence[3][3] = 
+  {
+    {1, 2, 0},
+    {0, 2, 1},
+    {0, 1, 2}
+  };
+  
+  const int
+  t8_edge_vertex_to_tree_vertex[T8_ECLASS_MAX_EDGES][2] =
+  {
+    {0, 1}, {2, 3}, {4, 5}, {6, 7}, {0, 2}, {4, 6}, {1, 3}, {5, 7}, {0, 4}, {1, 5}, {2, 6}, {3, 7}    /* hex */
+  };
+
+  double interpolated_coords[3], param[2], edge_delta[3]= {0, 0, 0}, cur_edge_delta[3], surface_deltas[6][3];
+  gp_Pnt pnt;
+  
+  /* Check each edge for geometry */
+  for (int i = 0; i < 12; ++i)
+  {
+    if (edges[i] >= 0)
+    {
+      /* Interpolate coordinates between edge vertices */
+      for (int j = 0; j < 3; ++j)
+      {
+        interpolated_coords[j] = (1 - ref_coords[i % 3]) * active_tree_vertices[t8_edge_vertex_to_tree_vertex[i][0] * 3 + j]
+                                + ref_coords[i % 3] * active_tree_vertices[t8_edge_vertex_to_tree_vertex[i][1] * 3 + j];
+      }
+
+      /* Interpolate parameters between edge vertices */
+      const double *parameters = (double *) t8_cmesh_get_attribute(cmesh, t8_get_package_id (),
+                                                                  T8_CMESH_OCC_CURVE_PARAMETERS_ATTRIBUTE_KEY + i,
+                                                                  gtreeid);
```

Comment:
This should be part of load_tree_data

Reply:
This is not possible, because we only load the parameters for the specific edge we are looking at. And we cannot load all edges at once, because if an edge is not linked there are no parameters and we get a memory error